lib/sqlalchemy/dialects/oracle/base.py

```diff
@@ -1658,8 +1658,8 @@ def get_columns(self, connection, table_name, schema=None, **kw):
         text = """
             SELECT col.column_name, col.data_type, col.%(char_length_col)s,
               col.data_precision, col.data_scale, col.nullable,
-              col.data_default, com.comments\
-            FROM all_tab_columns%(dblink)s col
+              col.data_default, com.comments, col.virtual_column\
+            FROM all_tab_cols%(dblink)s col
```

Comment:
I'm not sure if this has unintended consequences. 

From the documentation it should be ok:

> This view differs from "ALL_TAB_COLUMNS" in that hidden columns are not filtered out.

https://docs.oracle.com/cd/B19306_01/server.102/b14237/statviews_2093.htm#I1020276

Reply:
that might affect things, sure.      By "filtered out" they mean the previous view wont return as many rows, right?   it looks like this new view has a "HIDDEN_COLUMN" column which we could filter on by adding "hidden_column='NO'", which per this documentation would give us the same result rows.

is it the case that if we create a table in Oracle with a "Computed" on it, the *whole column* won't come back in reflection right now?

Reply:
no, by filtered out they mean columns. the column `virtual_column` is not present in `all_tab_columns`. The row with the computed column appears, since oracle returns the computed expression as the default, but there is not the column that indicates if it computed or not

https://docs.oracle.com/cd/B19306_01/server.102/b14237/statviews_2093.htm
> This view differs from "ALL_TAB_COLUMNS" in that hidden columns are not filtered out.

From the documentation the change should have no side effect

Reply:
OK just as long as there aren't rows coming back with "hidden=yes" that were rows we weren't getting before, it should be fine.

Reply:
per https://stackoverflow.com/questions/26790333/difference-between-sys-all-tab-columns-and-sys-all-tab-cols-in-oracle-12c the number of rows changes  "ALL_TAB_COLUMNS lists only user-defined columns whereas ALL_TAB_COLS also seems to contain some system columns in each user table that are not normally visible."   this will be backwards incompatible if we don't adjust by adding "hidden='NO'".  it seems like "hidden" and "invisible" are the two terms we need to deal with.

Reply:
I missed that part.. it does not seem to be listed in the documentation..
Should we add a where clause or join the two tables so we are sure that the returned rows arr the dame?

Reply:
I think if we add "where hidden='NO'" that should be enough.   the test suite will let us know.

Reply:
Will do. Sadly I cannot test against oracle

Reply:
no worries it'll happen on this end